tests/unit/Core/Service/ServiceSourceResolverTest.php

```diff
@@ -77,198 +76,400 @@ public function testSupportSelfManagedManifestsWithHttpUrls(): void
 
-        $metadata = Metadata::fromArray([
-            'name' => 'MyCoolService',
-            'label' => [],
-            'author' => 'Shopware',
-            'copyright' => 'Shopware',
-            'license' => 'Shopware',
-            'version' => '6.7.0.0',
-        ]);
+        $appInfo = new AppInfo(
+            'TestService',
+            '1.0.0',
+            'abc123',
+            '1.0.0-abc123',
+            'https://example.com/app.zip',
+            'sha256',
+            '6.6.0.0'
+        );
 
-        $manifest->method('getMetadata')->willReturn($metadata);
+        $result = $source->filesystemForVersion($appInfo);
 
-        yield 'manifest' => [$manifest];
+        static::assertSame('/tmp/test/TestService', $result->location);
     }
 
-    #[DataProvider('appProvider')]
-    public function testAppIsDownloadedIfItDoesNotExistOnFilesystem(AppEntity|Manifest $app): void
+    public function testFilesystemWhenAppExists(): void
     {
-        $serviceClientFactory = $this->createMock(ServiceClientFactory::class);
-
+        $client = $this->createMock(Client::class);
+        $temporaryDirectoryFactory = $this->createMock(TemporaryDirectoryFactory::class);
         $appExtractor = $this->createMock(AppExtractor::class);
         $filesystem = $this->createMock(Filesystem::class);
 
-        $appInfo = new AppInfo('MyCoolService', '6.7.0.0', 'abcd', '6.7.0.0-abcd', 'https://mycoolservice.com/service/lifecycle/app-zip/6.7.0.0');
+        $temporaryDirectoryFactory->expects($this->once())
+            ->method('path')
+            ->willReturn('/tmp/test');
+
+        $filesystem->expects($this->once())
+            ->method('exists')
+            ->with('/tmp/test/TestService')
+            ->willReturn(true);
 
-        $serviceClient = $this->createMock(ServiceClient::class);
-        $serviceClient->expects($this->once())->method('latestAppInfo')->willReturn($appInfo);
-        $serviceClient->expects($this->once())
-            ->method('downloadAppZipForVersion')
-            ->with('https://mycoolservice.com/service/lifecycle/app-zip/6.7.0.0', '/some/tmp/path/MyCoolService/MyCoolService.zip')
-            ->willReturn($appInfo);
-        $serviceClientFactory->expects($this->once())->method('fromName')->with('MyCoolService')->willReturn($serviceClient);
+        // Should not call download methods when app exists
+        $client->expects($this->never())->method('fetchServiceZipVersion');
+        $appExtractor->expects($this->never())->method('extract');
 
+        $source = new ServiceSourceResolver($client, $temporaryDirectoryFactory, $appExtractor, $filesystem);
+
+        $app = new AppEntity();
+        $app->setId(Uuid::randomHex());
+        $app->setName('TestService');
+        $app->setSourceType('service');
+
+        $result = $source->filesystem($app);
+
+        static::assertSame('/tmp/test/TestService', $result->location);
+    }
+
+    public function testFilesystemWhenAppDoesNotExist(): void
```

Comment:
Previous name was better imo, it actually describes what should happen: `testAppIsDownloadedIfItDoesNotExistOnFilesystem`